### PR TITLE
add batch_size check for max_pool2d_backward

### PR DESCRIPTION
**Pull request by jiayisunx**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141657

Fix https://github.com/pytorch/pytorch/issues/140923.

### CODE REVIEW

**Review by mingfeima — APPROVED** [review]

**Comment by jiayisunx** [comment]
hi @malfet  , could you please review this PR? thanks!

File: aten/src/ATen/native/Pool.h [hunk 1]
```diff
@@ -191,6 +191,12 @@ max_pool2d_backward_shape_check(
   check_dim_size(indices, ndim, ndim-3, nOutputPlane);
   check_dim_size(indices, ndim, ndim-2, outputHeight);
   check_dim_size(indices, ndim, ndim-1, outputWidth);
+
+  if (ndim == 4) {
+    const int64_t BatchSize = input.size(0);
```
Comment (malfet) [inline aten/src/ATen/native/Pool.h:196]: Please use `cameCase` (or actually `snake_case`)
```suggestion
    const int64_t batchSize = input.size(0);
```

File: aten/src/ATen/native/Pool.h [hunk 2]
```diff
@@ -191,6 +191,12 @@ max_pool2d_backward_shape_check(
...
+    check_dim_size(gradOutput, ndim, 0, BatchSize);
+    check_dim_size(indices, ndim, 0, BatchSize);
+  }
```
Comment (malfet) [inline aten/src/ATen/native/Pool.h:199]: Is there a check that dim is only 4 or 5?

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**